documentation/modules/exploit/windows/http/apache_activemq_traversal_upload.md

```diff
@@ -0,0 +1,91 @@
+## Introduction
+
+A directory traversal vulnerability was discovered in the fileserver upload/download functionality for blob messages in Apache ActiveMQ 5.x before 5.11.2 for Windows. The vulnerability, tracked as CVE-2015-1830, allows remote attackers to create JSP files in arbitrary directories via unspecified vectors.
+
+Because vulnerable servers allow for directory traversal, they will accept HTTP PUT requests for `/fileserver/..\\admin\\` and process these as requests for `/admin/`. For the PUT request to succeed, credentials need to be provided.
+
+This module exploits CVE-2015-1830 by attempting to upload a JSP payload to a target via an HTTP PUT requests for `/fileserver/..\\admin\\` using the default credentials `admin:admin` (or any other credentials provided by the user). It then issues an HTTP GET request to `/admin/<payload>.jsp` on the target in order to trigger the payload and obtain a shell. The module has been succesfully tested against ActiveMQ 5.11.1 on a Windows 7 machine.
+
+## Verification Steps
+
+1. Install the module as usual
+2. Start msfconsole
+3. Do: `use exploit/windows/http/apache_activemq_traversal_upload`
+4. Do: `set RHOSTS [IP]`
+5. Do: `set payload [payload]`
+6. Do: `set LHOST [IP]`
+7. Do: `exploit`
+
+## Options
+
+1.  `PASSWORD`. The default setting is `admin`.
+2.  `PATH`. This option is the traversal path. `/fileserver/..\admin\` by default.
+3.  `Proxies`. This option is not set by default.
+4.  `RHOSTS`. To use: `set RHOSTS [IP]`
+5.  `RPORT`. The default setting is `8161`. To use: `set RPORT [PORT]`
+6.  `SSL`. The default setting is `false`.
+7.  `THREADS`. The default setting is `1`.
+8.  `USERNAME`. The default setting is `admin`.
+9.  `VHOST`. This option is not set by default.
+10. `TARGETURI`. This option is the base path. `/` by default.
+
+## Compatible Payloads
+
+0. `generic/custom`
+1. `generic/shell_bind_tcp`
+2. `generic/shell_reverse_tcp`
+3. `java/jsp_shell_bind_tcp`
+4. `java/jsp_shell_reverse_tcp`
+
+## Payload Options
+1. `LHOST`. To use: `set LHOST [IP]`
+2. `LPORT`. The default setting is `4444`. To use: `set LPORT [PORT]`
+3. `SHELL`. This option is not set by default.
+
+## Scenarios
+
+```
```

Comment:
What is this scenario? What kind of machine/software version is the target running?